src/app/features/de-duplication/bulk-merge-records/bulk-merge-records.component.html

```diff
@@ -11,40 +11,15 @@
           <strong>{{ field.label }}</strong>
         </div>
 
-        @for (
-          entityToMerge of entitiesToMerge;
-          track entityToMerge;
-          let i = $index
-        ) {
-          <div class="record-cell">
-            @if (field.allowsMultiValueMerge) {
-              <mat-checkbox
-                [checked]="existingFieldSelected[field.id][i]"
-                (change)="selectExistingValue(field.id, i, $event.checked)"
-                [disabled]="isFieldDisabled[field.id][i]"
-              >
-                <app-entity-field-view
-                  [field]="field"
-                  [entity]="entityToMerge"
-                ></app-entity-field-view>
-              </mat-checkbox>
-            } @else {
-              <mat-radio-group>
-                <mat-radio-button
-                  [value]="entityToMerge[field.id]"
-                  (click)="selectExistingValue(field.id, i)"
-                  [checked]="existingFieldSelected[field.id][i]"
-                  [disabled]="isFieldDisabled[field.id][i]"
-                >
-                  <app-entity-field-view
-                    [field]="field"
-                    [entity]="entityToMerge"
-                  ></app-entity-field-view>
-                </mat-radio-button>
-              </mat-radio-group>
-            }
-          </div>
-        }
+        <app-merge-fields
+          [field]="field"
+          [entities]="entitiesToMerge"
+          [existingSelected]="existingFieldSelected[field.id]"
+          [isDisabled]="isFieldDisabled[field.id]"
+          [control]="mergeForm.formGroup.get(field.id)"
+          (valueChanged)="updateMergePreviewValue(field.id, $event)"
```

Comment:
I think existingSelected, isDisabled and valueChanged should be possible to handle (create and update) entirely inside the sub-component. Did you try that?

`existingFieldSelected` and `isFieldDisabled` as a status for all fields is probably not needed and each app-merge-fields component can internally track these just for its own field without inputs?